pkg/server/models.go

```diff
@@ -0,0 +1,24 @@
+package server
+
+type RecommendationData struct {
+	ClusterNamespace     map[string]string //ex: {"ns_local-cluster_ocm": 12343212}
+	RecommendationID     map[string]string // ex. {12343212: "ns_local-cluster_ocm" }
+	RecommendationStatus map[string]string
+	Recommendation       map[string][]map[string][]map[string]string
+}
```

Comment:
Why are all these maps?
I expect this object to capture a **single** recommendation instance.

If needed we could add a `RecommendationList` or `RecommendationStore` object to help us aggregate the individual instances.

Reply:
some of the requests will only have clusterid and namespace while some may have the recommendationid. Thats what we had before (recommendationList object to aggregate individual instance).

Reply:
I was thinking of something like this.
```go
type RecommendationItem struct {
	Cluster              string // Optional if we have recommentationID.
	Namespace            string // Optional if we have recommentationID.
	RecommendationID     string // Optional if we have Cluster+Namespace
	RecommendationStatus string
	Recommendation       map[string][]map[string]string
	// Eventually we can add more fields like time submitted, etc.
}

func NewRecommendation(...) *RecommendationItem{
  // This function will enforce that the RecommendationItem gets created with any required fields.
  return &RecommendationItem{}
}

type RecommendationStore struct
  data []*RecommendationItem
}

func (s *RecommendationStore) getById(recommendationID string) *RecommendationItem{
  // Use the ID to match the RecommendationItem
}

func  (s *RecommendationStore) getByClusterAndNamespace(cluster string, namespace string) *RecommendationItem{
  // Use the cluster and namespace to match the RecommendationItem
}
```